stubs/translationstring/translationstring/__init__.pyi

```diff
@@ -0,0 +1,81 @@
+from collections.abc import Callable
+from gettext import NullTranslations
+from typing import Any, Protocol
+from typing_extensions import Self
+
+class _TranslationStringFactory(Protocol):
+    def __call__(
+        self,
+        msgid: str | TranslationString,
+        mapping: dict[str, Any] | None = ...,
+        default: str | None = ...,
+        context: str | None = ...,
+    ) -> TranslationString: ...
+
+class _ChameleonTranslate(Protocol):
+    def __call__(
+        self,
+        msgid: str | TranslationString,
+        mapping: dict[str, Any] | None = ...,
+        context: str | None = ...,
+        target_language: str | None = ...,
+        default: str | None = ...,
+    ) -> TranslationString: ...
+
+class _TranslatorPolicy(Protocol):
+    def __call__(self, translations: NullTranslations, tstring: str, domain: str | None, context: str | None) -> str: ...
```

Comment:
As these are called using a positional call (i.e. implementors can call the arguments whatever they want to):

```suggestion
    def __call__(self, __translations: NullTranslations, __tstring: str, __domain: str | None, __context: str | None) -> str: ...
```

Reply:
True, the reason I did not do this, is, because that's what the docstring for the `policy` argument calls for, so technically Pylons could enforce the name of the parameters in a future revision, without breaking the contract laid out by the docs.